paddle/fluid/operators/math/concat_and_split.cu

```diff
@@ -144,12 +143,9 @@ class ConcatFunctor<platform::CUDADeviceContext, T> {
       }
       out_col += t_cols;
       inputs_col[i + 1] = out_col;
-      inputs_ptr[i] = const_cast<T*>(input[i].data<T>());
+      inputs_data[i] = const_cast<T*>(input[i].data<T>());
```

Comment:
this const_cast can be removed?

Reply:
Done in https://github.com/PaddlePaddle/Paddle/pull/15011.